### PR TITLE
Board

### PR DESCRIPTION
**Pull request by ahuvac**
- working on representing the tiles in the board
- adding methods to Player and Wallet

### CODE REVIEW

File: src/main/java/Board.java [hunk 1]
```diff
@@ -1,6 +1,40 @@
...
-    Hex[] board = new Hex[16];
+    //Hex[] board = new Hex[16];
+    List<Hex> board = new ArrayList<>();
+
+    public Board(){
+        for (int i = 0 ; i < 3; i++){
+            board.add( new Hex(ResourceType.ORE));
+        }
+
+        for (int i = 4 ; i < 7; i++){
```
Comment (wolfishl) [inline src/main/java/Board.java]: Do you need to do it that way? you can probably start at 0 for each

File: src/main/java/Hex.java [hunk 2]
```diff
@@ -1,10 +1,21 @@
...
+    public Hex(ResourceType type) {
         this.type = type;
-        this.position = position;
+        this.neighbors = new ArrayList();
+//        this.position = position;
+    }
+
+
+    public void connect(){
+
```
Comment (wolfishl) [inline src/main/java/Hex.java:18]: What is this for?

File: src/main/java/Player.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        developmentCards = new ArrayList<>();
+        cities = 4;
+        settlements = 5;
+        roads = 15;
+        score = 0;
+        numRoads = 0;
+    }
+
+    public void buildRoad(){
+        roads--;
```
Comment (wolfishl) [inline src/main/java/Player.java]: maybe only decrement after the player used cards so that if there aren't enough cards the roads isn't already decremented

File: src/main/java/Player.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        wallet.useCard(ResourceType.GRAIN);
+        wallet.useCard(ResourceType.GRAIN);
+        score+=2;
+    }
+
+    public void buyDevelopmentCard(){
+        wallet.useCard(ResourceType.GRAIN);
+        wallet.useCard(ResourceType.ORE);
+        wallet.useCard(ResourceType.WOOL);
+    }
```
Comment (wolfishl) [inline src/main/java/Player.java:86]: Add a card to developmentCards

File: src/main/java/Wallet.java [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+        cards.put(ResourceType.ROBBER, 0);
+        cards.put(ResourceType.WOOL, 0);
+    }
+
+    public void addCard(ResourceType card){
+        int old = cards.get(card);
+        cards.put(card, old++);
+    }
+
+    public void useCard(ResourceType card){
```
Comment (wolfishl) [inline src/main/java/Wallet.java]: What happens if old is 0? Also, the card should be returned to the bank when it is used

File: src/main/java/Player.java [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+public class Player {
+    private int settlements;
+    private int cities;
+    private int roads;
+    private int score;
+    private List<DevelopmentCard> developmentCards;
+    private Wallet wallet;
+    int numRoads;
+
```
Comment (wolfishl) [inline src/main/java/Player.java:14]: Should there be an arrayList of cities, settlements, and roads to keep track of their location or should that be part of Board?

File: src/main/java/Board.java (hunk 1 above)
Comment (ahuvac) [reply src/main/java/Board.java]: You're right. Did it that way because Nina had put in position, but i removed position in my pr

File: src/main/java/Hex.java (hunk 2 above)
Comment (ahuvac) [reply src/main/java/Hex.java:18]: to add neighbors, gonna be randomly assigned. Or makes more sense to randomly assign the resources to the tiles?

File: src/main/java/Player.java (hunk 3 above)
Comment (ahuvac) [reply src/main/java/Player.java]: true, I'll add to check if there is any left first

File: src/main/java/Wallet.java [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+
+    private Map<ResourceType, Integer> cards = new HashMap<>();;
+
+    public Wallet(){
+        cards.put(ResourceType.BRICK, 0);
+        cards.put(ResourceType.DESERT, 0);
+        cards.put(ResourceType.GRAIN, 0);
+        cards.put(ResourceType.ORE, 0);
+        cards.put(ResourceType.LUMBER, 0);
+        cards.put(ResourceType.ROBBER, 0);
```
Comment (nbildiri) [inline src/main/java/Wallet.java:12]: why is there a robber here?

File: src/main/java/Wallet.java [hunk 8]
```diff
@@ -0,0 +1,37 @@
+import java.util.*;
+
+public class Wallet {
+
+    private Map<ResourceType, Integer> cards = new HashMap<>();;
+
+    public Wallet(){
+        cards.put(ResourceType.BRICK, 0);
+        cards.put(ResourceType.DESERT, 0);
```
Comment (nbildiri) [inline src/main/java/Wallet.java]: why is this here?

File: src/main/java/Wallet.java [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+    public boolean useCards(List<ResourceType> cards){
+        // check if has each card
+//        if(cards.containsKey(card) && cards.get(card) == amount) {
+//            int num = cards.get(card);
+//            cards.put(card, num--);
+//            return true;
+//        }
+//        else
+            return false;
+    }
```
Comment (nbildiri) [inline src/main/java/Wallet.java:37]: need a method to check if over 7 cards in the wallet for the robber

File: src/main/java/Player.java (hunk 4 above)
Comment (nbildiri) [reply src/main/java/Player.java:86]: needs to take development card from the bank

Comment (ahuvac) [reply src/main/java/Player.java:86]: Can only add that in once Leahs code is merged
